File: mobile/android/core/session/src/main/kotlin/com/slopssaloon/omen/core/session/SecureSessionStore.kt
package com.slopssaloon.omen.core.session

/**
 * Abstraction over secure, at-rest session storage (M0c §2.2: "iOS Keychain / Android
 * Keystore-backed storage — never plain files, logs, or unencrypted preferences").
 *
 * The interface is pure Kotlin so the session/auth logic is JVM-unit-testable with an
 * in-memory fake. The production Android implementation ([AndroidKeystoreSessionStore])
 * encrypts the payload with an AES/GCM key held in the AndroidKeyStore.
 */
interface SecureSessionStore {
    /** Persist the session, encrypted at rest. Overwrites any existing session. */
    fun save(session: Session)

    /** Return the stored session, or null if none is present or it could not be decrypted. */
    fun load(): Session?

    /** Remove any stored session (sign-out / account deletion). */
    fun clear()
}

/**
 * Deterministic in-memory store for unit tests. NOT for production — provides no at-rest
 * encryption and is intentionally not Android-dependent.
 */
class InMemorySecureSessionStore(private var current: Session? = null) : SecureSessionStore {
    override fun save(session: Session) { current = session }
    override fun load(): Session? = current
    override fun clear() { current = null }
}

File: mobile/ios/OmenIOS/OmenIOS/Core/Session/SecureSessionStore.swift
import Foundation

/// Mirrors Android `core/session/SecureSessionStore.kt`. The production implementation is
/// `KeychainSessionStore`; `InMemorySecureSessionStore` is the deterministic fake used by tests
/// and SwiftUI previews.
protocol SecureSessionStore {
    func save(_ session: Session) throws
    func load() -> Session?
    func clear()
}

final class InMemorySecureSessionStore: SecureSessionStore {
    private var stored: Session?

    init(initial: Session? = nil) {
        stored = initial
    }

    func save(_ session: Session) throws {
        stored = session
    }

    func load() -> Session? {
        stored
    }

    func clear() {
        stored = nil
    }
}
